Framework/Geometry/inc/MantidGeometry/Instrument/ComponentInfo.h

```diff
@@ -54,19 +54,24 @@ class MANTID_GEOMETRY_DLL ComponentInfo {
   /// Pointer to the actual ComponentInfo object (non-wrapping part).
   std::unique_ptr<Beamline::ComponentInfo> m_componentInfo;
   /// Collection of component ids
-  boost::shared_ptr<const std::vector<Mantid::Geometry::IComponent *>>
-      m_componentIds;
+  boost::shared_ptr<const std::vector<Geometry::IComponent *>> m_componentIds;
   /// Map of component ids to indexes
   boost::shared_ptr<const std::unordered_map<Geometry::IComponent *, size_t>>
       m_compIDToIndex;
 
+  /// Shapes for each component
+  boost::shared_ptr<std::vector<boost::shared_ptr<const Geometry::Object>>>
+      m_shapes;
```

Comment:
@OwenArnold If you have instruments composed of repeating pixels, for example those defined using rectangular pixels, only one shape needs to be stored. Every other pixel is a copy of the original in another position. One only needs to know extents/relative rotations. Could that be a more compact/efficient way for some detector types. (Most instruments I have seen use repetition in the definition).